engine/src/main/scala/cromwell/engine/workflow/package.scala

```diff
@@ -0,0 +1,35 @@
+package cromwell.engine
+
+import akka.actor.{ActorSystem, ActorRef}
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.services.MetadataServiceActor.{MetadataServiceAction, PutMetadataAction, MetadataServiceResponse, MetadataEvent}
+
+import scala.concurrent.{Future, ExecutionContext}
+
+package object workflow {
+  // Sent when there is a failure to push to the metadata service
+  private[workflow] case class MetadataServiceFailureResponse(reason: Throwable)
+  private[workflow] case class MetadataActionFailureResponse(reason: Throwable, action: MetadataServiceAction)
+
+  object WorkflowMetadataKeys {
+    val Name = "workflowName"
+    val Id = "id"
+    val SubmissionTime = "submission"
+    val StartTime = "start"
+    val EndTime = "end"
+    val Status = "status"
+    val Outputs = "outputs"
+    val Inputs = "inputs"
+    val Calls = "calls"
+  }
+
+  implicit class EnhancedServiceRegistry(val serviceRegistryActor: ActorRef) extends AnyVal {
```

Comment:
I'm probably one of the most `implicit` friendly people in red team, but on this one, you're basically adding a `publishToMetadataService` method to any potential `ActorRef` in scope, which is a bit strange.

Maybe make it a trait instead ?

```scala
trait MetadataServiceUtil extends StrictLogging {
    def serviceRegistryActor: ActorRef

    def publishToMetadataService(message: MetadataEvent)(implicit actorSystem: ActorSystem, ec: ExecutionContext): Future[MetadataServiceResponse] = {
      import cromwell.util.PromiseActor._

      // Since this will always publish, we expect the callee to just provide an event,
      // and we wrap it into a `PutMetadataAction` before sending it to the MetadataService
      serviceRegistryActor.askNoTimeout(PutMetadataAction(message)).mapTo[MetadataServiceResponse].recover {
        case reason: Throwable =>
          // TODO: Do we still proceed with the execution? What do we do with the exception here?
          logger.warn(s"Failed to push event $message in the MetadataService: ${reason.getMessage}", reason)
          throw reason
      }
    }
  }
```